### PR TITLE
fix: dataset link by slug should have a fallback to any of the dataset with the slug shared with the workspace (HEXA-1367)

### PR DESCRIPTION
**Pull request by yolanfery**
Implement successive fallbacks for `datasetLinkBySlug`
Add unit tests to cover edge cases

### CODE REVIEW

File: backend/hexa/datasets/schema/queries.py [hunk 1]
```diff
@@ -64,17 +65,44 @@ def resolve_dataset_link(_, info, **kwargs):
...
+    if not workspace or not workspace.organization:
         return None
+    result = user_links.filter(
+        dataset__slug=dataset_slug,
+        dataset__shared_with_organization=True,
+        dataset__workspace__organization=workspace.organization,
+    ).first()
+    if result:
+        return result
+
```
Comment (nazarfil) [inline backend/hexa/datasets/schema/queries.py:104]: Good  priority based approach

**Review by nazarfil — APPROVED** [review]
